components/wifiManager/wifiManager/WiFiScanner.cpp

```diff
@@ -0,0 +1,205 @@
+                            all_records.push_back(ch_records[i]);
+                        }
+                    }
+                    delete[] ch_records;
+                }
+            }
+            vTaskDelay(pdMS_TO_TICKS(50));
+        }
+        
+        // Process all collected records
+        for (const auto& record : all_records) {
+            WiFiNetwork network;
+            network.ssid = std::string(reinterpret_cast<const char*>(record.ssid));
+            network.channel = record.primary;
+            network.rssi = record.rssi;
+            memcpy(network.mac, record.bssid, 6);
+            network.auth_mode = record.authmode;
+            scan_results.push_back(network);
+        }
+        
+        // Skip the normal result processing
+        return scan_results;
+    }
+    
+    // Wait for scan completion with timeout
+    int timeout_ms = 15000; // 15 second timeout
+    int elapsed_ms = 0;
+    
+    while (elapsed_ms < timeout_ms) {
+        uint16_t temp_count = 0;
+        esp_err_t count_err = esp_wifi_scan_get_ap_num(&temp_count);
+        
+        if (count_err == ESP_OK) {
+            // Wait a bit longer after finding networks to ensure scan is complete
+            if (temp_count > 0 && elapsed_ms > 5000) {
+                break;
+            }
+        }
+        
+        vTaskDelay(pdMS_TO_TICKS(200));
+        elapsed_ms += 200;
+    }
+    
+    if (elapsed_ms >= timeout_ms) {
+        ESP_LOGE(TAG, "Scan timeout after %d ms", timeout_ms);
+        esp_wifi_scan_stop();
+        return scan_results;
+    }
+    
+    // Get scan results
+    uint16_t ap_count = 0;
+    esp_wifi_scan_get_ap_num(&ap_count);
+    
+    if (ap_count == 0) {
+        ESP_LOGI(TAG, "No access points found");
+        return scan_results;
+    }
+
+    wifi_ap_record_t* ap_records = new wifi_ap_record_t[ap_count];
+    err = esp_wifi_scan_get_ap_records(&ap_count, ap_records);
+    if (err != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to get scan records: %s", esp_err_to_name(err));
+        delete[] ap_records;
+        return scan_results;
+    }
+
+    // Build the results vector and track channels found
+    bool channels_found[15] = {false}; // Track channels 0-14
+    
+    for (uint16_t i = 0; i < ap_count; i++) {
+        WiFiNetwork network;
+        network.ssid = std::string(reinterpret_cast<char*>(ap_records[i].ssid));
+        network.channel = ap_records[i].primary;
+        network.rssi = ap_records[i].rssi;
+        memcpy(network.mac, ap_records[i].bssid, 6);
+        network.auth_mode = ap_records[i].authmode;
+        
+        if (network.channel <= 14) {
+            channels_found[network.channel] = true;
+        }
+        
+        scan_results.push_back(network);
+    }
+    
+
+    delete[] ap_records;
+    ESP_LOGI(TAG, "Found %d access points", ap_count);
+    
+    // Also update the member variable for compatibility
+    networks = scan_results;
```

Comment:
We don't have a method to grab the stored networks, can we maybe refactor the thing so that we don't need to needlesly store them?